mne_sandbox/decoding/_time_frequency_csp.py

```diff
@@ -0,0 +1,255 @@
+        signals.
+    tmax : float
+        Times in seconds. Upper bound of the time over which to estimate
+        signals.
+    freqs : array of floats
+        Frequency values over which to compute time-frequency decomposition.
+    estimator : object
+        Scikit-learn classifier object.
+    sfreq : float
+        Sample frequency of data.
+    n_cycles : float, defaults to 7.
+        Number of cycles that each frequency should complete in forming the
+        sliding temporal window.
+        The time-window length is T = n_cycles / freq.
+    scorer : object | None | str
+        scikit-learn Scorer instance or str type indicating the name of the
+        scorer such as ``accuracy``, ``roc_auc``. If None, set to ``accuracy``.
+    n_jobs : int
+        Number of jobs to run in parallel. Defaults to 1.
+
+    Attributes
+    ----------
+    ``filters_`` : ndarray, shape (n_channels, n_channels)
+        If fit, the CSP components used to decompose the data, else None.
+    ``patterns_`` : ndarray, shape (n_channels, n_channels)
+        If fit, the CSP patterns used to restore M/EEG signals, else None.
+    ``mean_`` : ndarray, shape (n_components,)
+        If fit, the mean squared power for each component.
+    ``std_`` : ndarray, shape (n_components,)
+        If fit, the std squared power for each component.
+    """
+
+    def __init__(self, tmin, tmax, freqs, estimator, sfreq, n_cycles=7.,
+                 scorer=None, n_jobs=1):
+        """Init of TimeFrequencyCSP."""
+
+        # check frequency list
+        if len(freqs) < 2:
+            raise ValueError('freqs must contain more than one value.')
+        self.freqs = freqs
+
+        # check sfreq
+        if not isinstance(sfreq, (float, int)):
+            raise ValueError('sfreq must be a float or an int, got %s '
+                             'instead.' % type(sfreq))
+        self.sfreq = float(sfreq)
+
+        # check n_cycles
+        if not isinstance(n_cycles, (float, int)):
+            raise ValueError('sfreq must be a float or an int, got %s '
+                             'instead.' % type(n_cycles))
+        self.n_cycles = float(n_cycles)
+
+        # TODO:add test that the number of cycles is appropriate for epochs
+
+        self.estimator = estimator
+        self.scorer = scorer
+        self.n_jobs = n_jobs
+
+        # Assemble list of frequency range tuples
+        self._freq_ranges = zip(self.freqs[:-1], freqs[1:])
+
+        # Infer window spacing from the max freq and n cycles to avoid gaps
+        self._window_spacing = (n_cycles / np.max(freqs) / 2.)
+        self._centered_w_times = np.arange(tmin + self._window_spacing,
+                                           tmax - self._window_spacing,
+                                           self._window_spacing)[1:]  # noqa
+        self._n_windows = len(self._centered_w_times)
+
+    def _transform(self, epochs, y, w_tmin, w_tmax, method='', pos=[]):
```

Comment:
Why do the `method` and `pos` params have default values? The defaults don't make much sense to me.